server.js

```diff
@@ -93,34 +88,32 @@ app.post("/signup", async (req, res) => {
 })
 
 app.post("/login", async (req, res) => {
-    const loginuser = await User.findOne({ username: req.body.username })
     try {
-        if (loginuser) {
+        const loginuser = await User.findOne({ username: req.body.username })
+        if (!loginuser) {
+            res.redirect('/');
+        } else {
             const validate = await bcrypt.compare(req.body.password, loginuser.password)
-            if (validate) {
+            if (!validate) {
+                res.redirect('/');
+            } else {
                 const newSession = await Session.create({
-                    user: req.body.username,
+                    user: loginuser.id,
                 });
-                res.cookie('session_id', newSession.id);
-                console.log(loginuser.id);
-                res.redirect(`/profile/${loginuser.id}`);
-            } else {
+                res.cookie('session', newSession);
```

Comment:
So you give me, the user, the full session? What's stopping me from editing it and putting whatever user ID I want there?

The idea is that the user only gets the session ID, and only the server knows what kind of data is associated with each session. Else the user can simply change their cookie to say "I am admin" and your code would simply believe that to be true.

Reply:
I agree with everything else (staga dush) but the thing that's stopping you from editig it is the database. Only exisiting users can reach the profile page

Reply:
Cool, so I can just change the userID in my session object and I can be any other user, no need for their password :D

Reply:
RIP